### PR TITLE
[feature] Pivot: column drilldown

### PR DESCRIPTION
**Pull request by re1ro**
[GS-2136](https://govspend.atlassian.net/browse/GS-2136)
* [x] Pivot pagination: introducing drilldown for rows and columns
* [x] Unit tests for column `drilldown` and `expanded` reactor

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>should support onDispatch (and pivot resetting drilldown)</summary>
  <code>expected [ 'Florida' ] to deeply equal []</code>
</details>
<details>
  <summary>should support onDispatch (and pivot resetting drilldown)</summary>
  <code>expected [  ] to deeply equal []</code>
</details>
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 05ffdd60c5a23d55cc1ca7104852f6c8ff3a5ff9


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5f1bcd74022af600938e2fbd858ecad3c7c0dae7
</p>

**Comment by ccalbaugh** [comment]
@re1ro Are we still waiting on more tests for this one?

**Review by ccalbaugh — COMMENTED** [review]
Looks great so far!  Just waiting on tests and conflict resolution.

File: src/pivot.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  },
+  // Resetting the pagination when the tree is changed
+  // allows to return expected root results instead of nested drilldown
+  // EX: criteria filters didn't work properly when drilldown was applied
+  onUpdateByOthers(node, extend) {
+    resetPagination(extend, node, node)
+  },
+  shouldMergeResponse: node =>
+    // checking for presence of drilldown, skip, expanded in pagination
+    someNotEmpty(
```
Comment (dshishkov) [inline src/pivot.js:196]: comment here mentions `expanded` but I don't see it as part of the check

Comment (ZavoiuDaniel) [reply src/pivot.js]: There appears to be an expand method in [src/exampleTypes/pivot.js](https://github.com/smartprocure/contexture-client/pull/186/files#diff-7135becdbd16d012b9e6f46a692c49b112fadcfe849d74f8a8610c482496eca1), uncertain if it's related to the same item.

Comment (re1ro) [reply src/pivot.js]: Yeah, this structure is changed entirely now. Removed the comment

**Review by stellarhoof — APPROVED** [review]

File: test/index.js [hunk 2]
```diff
@@ -2106,6 +2106,7 @@ let AllTests = ContextureClient => {
     )
     let node = Tree.getNode(['root', 'pivot'])
 
+    // TODO test the loaded field population
```
Comment (ccalbaugh) [inline test/index.js:2109]: Do we still need to do this in this PR?

File: src/exampleTypes/pivot.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
+import _ from 'lodash/fp'
+import { toJS } from 'mobx'
+
+let getKey = x => x.keyAsString || x.key
```
[the remark this answers is not in the document]
Comment (ccalbaugh) [reply src/exampleTypes/pivot.js:4]: or `x?.keyAsString ?? x?.key ?? null` if you prefer the whole nullish coalescing and optional chaining thing

**Review by dshishkov — APPROVED** [review]

**Comment by dshishkov** [comment]
@re1ro are any of the remaining comments from @ccalbaugh worth addressing before merging this?

**Review by ccalbaugh — APPROVED** [review]

File: src/exampleTypes/pivot.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ccalbaugh) [reply src/exampleTypes/pivot.js:4]: Per a conversation with @re1ro `x` should never be undefined, we should treat it as an error if so.

File: test/index.js (hunk 2 above)
Comment (re1ro) [reply test/index.js:2109]: Yeah I will do more work toward it separately

**Pull request merged**